rs/client-gen/src/lib.rs

```diff
@@ -7,58 +13,106 @@ mod root_generator;
 mod service_generators;
 mod type_generators;
 
-use root_generator::RootGenerator;
+const SAILS: &str = "sails_rs";
 
-use anyhow::{Context, Result};
-use convert_case::{Case, Casing};
-use sails_idl_parser::ast::{visitor, Program};
-use std::{ffi::OsStr, fs, io::Write, path::Path};
+pub struct IdlPath<'a>(&'a Path);
+pub struct IdlString<'a>(&'a str);
+pub struct ClientGenerator<'a, S> {
+    sails_path: Option<&'a str>,
+    mocks_feature_name: Option<&'a str>,
+    idl: S,
+}
+
+impl<'a, S> ClientGenerator<'a, S> {
+    pub fn with_mocks(self, mocks_feature_name: &'a str) -> Self {
+        Self {
+            mocks_feature_name: Some(mocks_feature_name),
+            ..self
+        }
+    }
+
+    pub fn with_sails_crate(self, sails_path: &'a str) -> Self {
+        Self {
+            sails_path: Some(sails_path),
+            ..self
+        }
+    }
+}
 
-pub fn generate_client_from_idl(
```

Comment:
We want to keep either this or renamed function as a shortcut so people don't need to work with the generator directly when they don't need any of optional params

Reply:
Leave it as it is for now